vpd-manager/manager.cpp

```diff
@@ -632,13 +632,52 @@ void Manager::deleteFRUVPD(const sdbusplus::message::object_path path)
 
     inventory::Path& vpdFilePath = std::get<0>(frus.find(objPath)->second);
 
-    string chipAddress =
-        jsonFile["frus"][vpdFilePath].at(0).value("pcaChipAddress", "");
+    // Implies the api has been called as a part of everest base panel CM
+    if (path.filename() == "panel0")
+    {
+        const std::vector<nlohmann::json>& groupEEPROM =
+            jsonFile["frus"][vpdFilePath]
+                .get_ref<const nlohmann::json::array_t&>();
+
+        const nlohmann::json& singleFru = groupEEPROM[0];
+
+        // check if the device qualifies for CM.
+        if (!singleFru.value("concurrentlyMaintainable", false))
+        {
+            elog<InvalidArgument>(Argument::ARGUMENT_NAME(
+                                      "Device not marked for CM, Object Path"),
+                                  Argument::ARGUMENT_VALUE(objPath.c_str()));
+            return;
+        }
+
+        if ((singleFru.find("devAddress") == singleFru.end()) ||
+            (singleFru.find("driverType") == singleFru.end()) ||
+            (singleFru.find("busType") == singleFru.end()))
+        {
+            // The FRUs is marked for collection but missing mandatory
+            // fields for CM. Log error and return.
+            log<level::ERR>(
+                "CM Failed as mandatory field missing in Json",
+                entry("ERROR=%s", ("CM failed for " + (objPath)).c_str()));
+
+            return;
+        }
 
-    // Unbind the LED driver for this FRU
-    cout << "Unbinding device- " << chipAddress << endl;
-    executeCmd(createBindUnbindDriverCmnd(chipAddress, "i2c", "leds-pca955x",
-                                          "/unbind"));
+        executeCmd(createBindUnbindDriverCmnd(
```

Comment:
Why is this even needed? What does unbinding the eeprom driver achieve?

Reply:
okay, I will remove this.

Reply:
> okay, I will remove this.

That was more of a question :) Do things work fine (such as VPD collected for the new part after replug) even if we don't unbind/bind?

Reply:
Actually, I removed the unbind part and tested, it is working fine. That is why I told that I will remove it.